src/sentry/static/sentry/app/views/eventsV2/table/tableView.tsx

```diff
@@ -83,22 +106,57 @@ class TableView extends React.Component<TableViewProps> {
    * Please read the comment on `_createColumn`
    */
   _updateColumn = (columnIndex: number, nextColumn: TableColumn<keyof TableDataRow>) => {
-    const {location, eventView, tableData} = this.props;
+    const {location, eventView, tableData, organization} = this.props;
 
     if (!tableData || !tableData.meta) {
       return;
     }
 
+    const payload = {
+      aggregation: String(nextColumn.aggregation),
+      field: String(nextColumn.field),
+      fieldname: nextColumn.name,
+    };
+
     const nextEventView = eventView.withUpdatedColumn(
       columnIndex,
-      {
-        aggregation: String(nextColumn.aggregation),
-        field: String(nextColumn.field),
-        fieldname: nextColumn.name,
-      },
+      payload,
       tableData.meta
     );
 
+    if (nextEventView !== eventView) {
+      const changed: string[] = [];
+
+      const prevField = explodeField(eventView.fields[columnIndex]);
+      const nextField = explodeField(nextEventView.fields[columnIndex]);
+
+      const aggregationChanged = prevField.aggregation !== nextField.aggregation;
+      const fieldChanged = prevField.field !== nextField.field;
+      const fieldnameChanged = prevField.fieldname !== nextField.fieldname;
+
+      if (aggregationChanged) {
+        changed.push('aggregate');
+      }
+
+      if (fieldChanged) {
+        changed.push('field');
+      }
+
+      if (fieldnameChanged) {
+        changed.push('fieldname');
+      }
+
+      // metrics
+      trackAnalyticsEvent({
+        eventKey: 'discover_v2.update_column',
+        eventName: 'Discoverv2: A column was updated',
+        updated_at_index: columnIndex,
+        changed,
+        organization_id: organization.id,
+        ...payload,
+      });
```

Comment:
It might make sense to track the column aggregation and field before/after the update.

Reply:
I'll add this in a follow up PR. I want to try to merge this in soon.